### PR TITLE
[feat]add metrics

### PR DESCRIPTION
**Pull request by hunshenshi**
add `w3b_receive_event_metrics` and `w3b_blockchain_tx_metrics` metrics

### CODE REVIEW

File: pkg/modules/vm/wasmtime/instance_exports.go [hunk 1]
```diff
@@ -309,6 +320,9 @@ func (ef *ExportFuncs) SendTX(chainID int32, offset, size, vmAddrPtr, vmSizePtr
 		ef.log.Error(err)
 		return wasm.ResultStatusCode_Failed
 	}
+
+	_blockChainTxMtc.WithLabelValues(types.MustProjectFromContext(ef.ctx).Name).Inc()
```
Comment (Liuhaai) [inline pkg/modules/vm/wasmtime/instance_exports.go:324]: move it into `ChainClient`

**Comment by Liuhaai** [comment]
any db metrics?

**Comment by hunshenshi** [comment]
> any db metrics?

from db metadata

File: pkg/modules/vm/wasmtime/instance_exports.go (hunk 1 above)
Comment (hunshenshi) [reply pkg/modules/vm/wasmtime/instance_exports.go]: fixed

File: pkg/types/wasm/wasm_config_chain_client.go [hunk 2]
```diff
@@ -59,7 +69,7 @@ func decodeEndpoints(in string) (ret map[uint32]string) {
 	return
 }
 
-func (c *ChainClient) SendTX(chainID uint32, toStr, valueStr, dataStr string) (string, error) {
+func (c *ChainClient) SendTX(projectName string, chainID uint32, toStr, valueStr, dataStr string) (string, error) {
```
Comment (Liuhaai) [inline pkg/types/wasm/wasm_config_chain_client.go:72]: the project object could be added under the ChainClient struct when created

**Review by CoderZhi — APPROVED** [review]

File: docker-compose.yaml [hunk 3]
```diff
@@ -82,3 +81,30 @@ services:
...
+    volumes:
+      - ./build/var/prometheus/conf/prometheus.yml:/etc/prometheus/prometheus.yml
+      - ${WS_WORKING_DIR:-.}/prometheus:/prometheus
+    ports:
+      - "9090:9090"
+#    environment:
+#      W3B_SERVICE: http://w3bapp:8888
+#      PG_ADAPTER_SERVICE: http://pg-prometheus:9201
+
+  pg-prometheus:
```
Comment (Liuhaai) [inline docker-compose.yaml:103]: The storage of the prometheus can be configured in the yaml directly? https://prometheus.io/docs/prometheus/latest/storage/#local-storage

File: pkg/modules/deploy/instance_context.go [hunk 4]
```diff
@@ -53,7 +54,7 @@ func WithInstanceRuntimeContext(parent context.Context) (context.Context, error)
 	if _, ok := wasm.KVStoreFromContext(ctx); !ok {
 		ctx = wasm.DefaultCache().WithContext(ctx)
 	}
-	ctx = wasm.WithChainClient(ctx, wasm.NewChainClient(parent))
+	ctx = wasm.WithChainClient(ctx, wasm.NewChainClient(ctx))
```
Comment (Liuhaai) [inline pkg/modules/deploy/instance_context.go:57]: NewChainClient(parent, prj, ... )

File: docker-compose.yaml (hunk 3 above)
Comment (hunshenshi) [reply docker-compose.yaml:103]: We use remote storage, so can't configure in the yaml file.

File: pkg/modules/deploy/instance_context.go (hunk 4 above)
Comment (hunshenshi) [reply pkg/modules/deploy/instance_context.go:57]: I think the project structure has memory in the context, so just get it from the context, not from a parameter.

File: pkg/depends/conf/http/server.go [hunk 5]
```diff
@@ -76,11 +76,12 @@ func (s *Server) Serve(router *kit.Router) error {
 	ht := s.ht
 	ht.Port = s.Port
 
-	ht.Middlewares = []httptransport.HttpMiddleware{mws.DefaultCompress}
+	ht.Middlewares = []httptransport.HttpMiddleware{}
 	ht.Middlewares = append(ht.Middlewares, middlewares...)
 	ht.Middlewares = append(ht.Middlewares,
 		mws.DefaultCORS(),
 		mws.HealthCheckHandler(),
+		mws.MetricsHandler(),
```
Comment (saitofun) [inline pkg/depends/conf/http/server.go:84]: should MetricsHandler be a optional handler?

Comment (hunshenshi) [reply pkg/depends/conf/http/server.go:84]: Metrics should be an essential component, right?

**Pull request merged**